File: src/test/java/bad/robot/radiate/ui/ExampleOvertimeIndicator.java
package bad.robot.radiate.ui;

import bad.robot.radiate.Progress;
import bad.robot.radiate.ui.swing.Debug;

import javax.swing.*;
import java.awt.*;

import static bad.robot.radiate.Activity.Progressing;
import static bad.robot.radiate.ui.swing.Debug.drawCentreLines;
import static javax.swing.WindowConstants.EXIT_ON_CLOSE;

public class ExampleOvertimeIndicator {

    public static void main(String[] args) throws InterruptedException {
        OvertimeIndicator indicator = setupWindow();
        indicator.setVisiblityBasedOn(Progressing, new Progress(100, 100));
    }

    private static OvertimeIndicator setupWindow() {
        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(EXIT_ON_CLOSE);
        frame.setSize(400, 400);
        JPanel panel = new JPanel() {
            @Override
            public void paint(Graphics g) {
                super.paint(g);
                Graphics2D graphics = (Graphics2D) g.create();
                drawCentreLines(this.getBounds(), graphics);
                graphics.dispose();
            }
        };
        panel.setBackground(Color.lightGray);
        OvertimeIndicator indicator = new OvertimeIndicator();
        frame.add(new JLayer<>(panel, indicator));
        frame.setVisible(true);
        return indicator;
    }

}

File: src/test/scala/bad/robot/radiate/ui/ExampleOvertimeIndicator.scala
package bad.robot.radiate.ui

import java.awt.{Color, Graphics, Graphics2D}
import javax.swing.WindowConstants._
import javax.swing.{JComponent, JFrame, JLayer, JPanel}

import bad.robot.radiate.ui.swing.Debug._
import bad.robot.radiate.Progress
import bad.robot.radiate.activity.Progressing

object ExampleOvertimeIndicator extends App {

  val indicator = setupWindow
  indicator.setVisibilityBasedOn(Progressing, new Progress(100, 100))

  private def setupWindow: OvertimeIndicator = {
    val frame = new JFrame
    frame.setDefaultCloseOperation(EXIT_ON_CLOSE)
    frame.setSize(400, 400)
    val panel = new JPanel() {
      override def paint(g: Graphics) {
        super.paint(g)
        val graphics = g.create.asInstanceOf[Graphics2D]
        drawCentreLines(this.getBounds, graphics)
        graphics.dispose()
      }
    }
    panel.setBackground(Color.lightGray)

    val indicator = new OvertimeIndicator
    frame.add(new JLayer[JComponent](panel, indicator))
    frame.setVisible(true)
    indicator
  }
}
